docs/design/fused_moe_modular_kernel.md

```diff
@@ -0,0 +1,226 @@
+# Fused MoE Modular Kernel
+
+## Introduction
+FusedMoEModularKernel is implemented [here](https://github.com/vllm-project/vllm/blob/main/vllm/model_executor/layers/fused_moe/modular_kernel.py)
+
+Based on the format of the input activations, FusedMoE implementations are broadly classified into 2 types.
+* Contiguous / Standard / Non-Batched, and
+* Batched
+
+*Note that the terms Contiguous, Standard, and Non-Batched are used interchangeably throughout the document.*
```

Comment:
```suggestion
!!! note
    The terms Contiguous, Standard, and Non-Batched are used interchangeably throughout the document.
```

Reply:
`!!! note` is mkdocs syntax. It's where the blue note boxes come from in the docs

Reply:
Outsite of the docs it does not look as good. Maybe one day we'll add something to parse [GitHub style admonitions](https://github.com/orgs/community/discussions/16925) into Mkdocs style ones in the docs build so we get the best of both